tests/compile-fail/super_trait_not_implemented.rs

```diff
@@ -0,0 +1,19 @@
+use auto_impl::auto_impl;
+
+trait Supi {}
+
+#[auto_impl(Box, &)]
+trait Foo: Supi {}
+
+
+struct Dog;
+impl Supi for Dog {}
+impl Foo for Dog {}
+
+
+fn requires_foo<T: Foo>(_: T) {}
+
+fn main() {
+    requires_foo(Dog); // should work
+    requires_foo(Box::new(Dog)); // shouldn't, because `Box<Dog>: Supi` is not satisfied
```

Comment:
Supertraits kind of break down here when you expect them to work like inheritance. You don't end up inheriting and extending behaviour, you end up with additional constraints to satisfy.

Sorry, this isn't a comment on this PR specifically, just an observation inspired by it :)